### PR TITLE
[IMP] filters: modernize look

### PR DESCRIPTION
**Pull request by LucasLefevre**
## Description:

Convert the data filter popover to the new style.

Task: [4358098](https://www.odoo.com/odoo/2328/tasks/4358098)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/components/filters/filter_menu/filter_menu.ts [hunk 1]
```diff
@@ -53,13 +42,6 @@ const CSS = css/* scss */ `
...
-
-      .o-filter-menu-action-text {
-        cursor: pointer;
-        margin-right: 10px;
-        color: blue;
-        text-decoration: underline;
-      }
     }
 
     .o-filter-menu-list {
```
Comment (hokolomopo) [inline src/components/filters/filter_menu/filter_menu.ts:47]: All the borders of the side panel have the same gray IIRC (GRAY_300 I think ?). Maybe this one should too

File: src/components/filters/filter_menu/filter_menu.xml [hunk 2]
```diff
@@ -3,22 +3,26 @@
...
+          <div class="o-filter-menu-item my-2" t-on-click="() => this.sortFilterZone('descending')">
             Sort descending (Z ⟶ A)
           </div>
         </div>
         <div class="o-separator"/>
       </t>
       <div class="o-filter-menu-actions">
-        <div class="o-filter-menu-action-text" t-on-click="selectAll">Select all</div>
-        <div class="o-filter-menu-action-text" t-on-click="clearAll">Clear</div>
+        <div class="o-button-link me-2 text-decoration-underline" t-on-click="selectAll">
```
Comment (hokolomopo) [inline src/components/filters/filter_menu/filter_menu.xml:18]: o-button-link are not underlined in the side panel. Idk if it fits here or not

File: src/components/side_panel/components/checkbox/checkbox.xml [hunk 3]
```diff
@@ -2,6 +2,7 @@
   <t t-name="o-spreadsheet.Checkbox">
     <label
       class="o-checkbox d-flex align-items-center"
+      role="button"
```
Comment (hokolomopo) [inline src/components/side_panel/components/checkbox/checkbox.xml:5]: image

Why is it now necessary ? :eyes:

File: src/components/filters/filter_menu_item/filter_menu_value_item.xml [hunk 4]
```diff
@@ -1,20 +1,22 @@
...
-          <span t-if="this.props.isChecked">✓</span>
-        </div>
-      </div>
-      <div class="o-filter-menu-value-text text-truncate">
+      <t t-set="value">
         <t t-if="this.props.value === ''">(Blanks)</t>
         <t t-else="" t-esc="this.props.value"/>
-      </div>
+      </t>
+      <!-- toString because t-set with a body creates a LazyValue instead of a string -->
```
Comment (hokolomopo) [inline src/components/filters/filter_menu_item/filter_menu_value_item.xml:12]: Kinda sucks that the comment will be in the DOM :disappointed:

**Review by hokolomopo — COMMENTED** [review]
Prettier thingies :raised_hands:

File: src/components/filters/filter_menu/filter_menu.xml (hunk 2 above)
Comment (LucasLefevre) [reply src/components/filters/filter_menu/filter_menu.xml:18]: I agree that underline isn't the prettiest. But without the underline, you see "Select all Clear" without any clear separation between the two buttons. I didn't want to completely reinvent how it looked. I'll add a not to François if he wants to improve it.

File: src/components/side_panel/components/checkbox/checkbox.xml (hunk 3 above)
Comment (LucasLefevre) [reply src/components/side_panel/components/checkbox/checkbox.xml:5]: To change the pointer to indicate that the label is clickable

Comment (rrahir) [reply src/components/side_panel/components/checkbox/checkbox.xml:5]: I edited the commit to have the cursor: pointer also applying to the input (some other rules interfered with it, namely the appearance:none and its variants)

**Comment by rrahir** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**